spec/models/manageiq/providers/openstack/cloud_manager/refresh_spec_common.rb

```diff
@@ -863,5 +863,107 @@ def expect_sync_cloud_tenants_with_tenants_is_queued
       expect(sync_cloud_tenant.method_name).to eq("sync_cloud_tenants_with_tenants")
       expect(sync_cloud_tenant.state).to eq(MiqQueue::STATE_READY)
     end
+
+    def assert_targeted_vm(vm_name, attributes)
```

Comment:
So for AWS, every time there is a reference to Vm (aka. Vm target), we unpack the list of references by looking at the DB:

https://github.com/Ladas/manageiq-providers-amazon/blob/6938025f2cd6a4b0504909a1be22a87e30dbc7c0/app/models/manageiq/providers/amazon/inventory/collector/target_collection.rb#L209

and by looking at the API:
https://github.com/Ladas/manageiq-providers-amazon/blob/6938025f2cd6a4b0504909a1be22a87e30dbc7c0/app/models/manageiq/providers/amazon/inventory/collector/target_collection.rb#L229

So we basically send more Targetsto the refresh, by scanning edges of the Vm.

-------

Then the Vm full refresh means that after 1 targeted refresh, you will get a full info about the Vm. Where the most important relations are probably IP addresses and volumes. Then it's easier to use this e.g. in a automate workflow, where 1 step is the sync refresh of the Vm and you want to work with the data in the next step.

---

Also this solves the fact that events might come in a wrong order, then the graph refresh will not be able to populate some edges. (for the 5.0 release, we will be solving these by a skeletal pre-create and a post processing where the missing edges will be planned for the next refresh)